nvflare/app_common/ccwf/client_ctl.py

```diff
@@ -357,15 +357,17 @@ def execute(self, task_name: str, shareable: Shareable, fl_ctx: FLContext, abort
                 message_handle_func=self._process_end_workflow,
             )
 
+            learnable = self.persistor.load(fl_ctx)
+            fl_ctx.set_prop(AppConstants.GLOBAL_MODEL, learnable, private=True, sticky=True)
+
```

Comment:
Why do we move this from "start_task_name" to "configure_task_name"?

Reply:
start_task_name is only for the starting client, and the persistor needs to be loaded to initialize the persistence_manager on all clients

Reply:
got it, thanks for the explanation.

One question here:

1. If all the clients want to start from the same initialized weights (to avoid non-convergence), are we doing this?
2. Or are we saying each of the client start with its own initialized weights ( which means each client can have a different starting point)

Reply:
No. Only the final model is brodcast at the end of the training. There is no need to broadcast the initial model, since it is only used by the starting client.

The reason that the initial model needs to be loaded here is because some model persistors won't work later when saving result if it didn't load an initial model.

Reply:
Currently, the starting client does send the initial model weights via _scatter to all the clients:
swarm_client_ctl.py:386 `return self.send_learn_task(targets=targets, request=task_data, fl_ctx=fl_ctx)`
where task_data is the initial_model model weights from client_ctl.py:372`return self.start_workflow(initial_model, fl_ctx, abort_signal)`

Reply:
So if I understand correctly:


At the very beginning,
All clients (c1, c2, c3, c4) after the "configure task" will set this "AppConstants.GLOBAL_MODEL" to what is loaded using the persistor (which is randomly initialized in pt's case) (m1, m2, m3, m4)

Then server will send "start_task" to the starting client (let's say c1), and the starting client (c1) will get this "AppConstants.GLOBAL_MODEL" as learnable (m1) and then BROADCAST to all participating training clients.

Reply:
Correct in the case of swarm learning. In case of cyclic, the starting client will train, and then sends its result to the next client, and so on.

Reply:
thanks for all discussions/explanations, I am much clearer.